# The REG301_EVENT_DATA Table

Welcome to an in-depth exploration of the ``REG301_EVENT_DATA`` table. This table captures important legally relevant actions that take place throughout the lifecycle of a patent application, primarily up to the end of the opposition period. It serves as a logbook, tracking various legal events and providing valuable insights into the procedural history of patent applications.

The ``REG301_EVENT_DATA`` table includes data such as event dates, event codes, and bulletin references, offering a comprehensive record of the significant steps in the patent granting process. Notably, the table also covers annual payments during the European Patent Office (EPO) patent grant procedure, including the opposition period. For a more extensive overview of legal events beyond this period, additional data from EPO member states can be found in the INPADOC legal events table (``TLS231_INPADOC_LEGAL_EVENT``).

In [1]:
from epo.tipdata.patstat import PatstatClient
from epo.tipdata.patstat.database.models import REG301_EVENT_DATA, REG101_APPLN
from sqlalchemy import select, func, case, select, and_

patstat = PatstatClient(env='TEST')

db = patstat.orm()

In [3]:

q = db.query(
    REG301_EVENT_DATA.id,
    REG301_EVENT_DATA.event_date,
    REG301_EVENT_DATA.event_code,
    REG301_EVENT_DATA.bulletin_year,
    REG301_EVENT_DATA.bulletin_nr,
    REG301_EVENT_DATA.bulletin_date
)

res = patstat.df(q)
res


id  event_date       event_code  bulletin_year  bulletin_nr  \
0         105615  2002-10-25          0009182           2002           50   
1         105463  2004-03-19          0009013           2004           19   
2        6255440  2007-10-05      0009199INVT           2007           45   
3       99941416  2007-10-05      0009299OPPO           2007           45   
4        7021392  2012-08-31          0009260           2012           40   
...          ...         ...              ...            ...          ...   
309541  16179719  2018-08-25  EPIDOSNPFE2RFEE              0            0   
309542  16179747  2018-09-05  EPIDOSNPFE2RFEE              0            0   
309543  16180553  2018-09-05  EPIDOSNPFE2RFEE              0            0   
309544  16180556  2018-08-25  EPIDOSNPFE2RFEE              0            0   
309545  16181247  2018-09-05  EPIDOSNPFE2RFEE              0            0   

       bulletin_date  
0         2002-12-11  
1         2004-05-06  
2         2007-11-07  
3         2007-11-07  
4         2012-10-03  
...              ...  
309541    9999-12-31  
309542    9999-12-31  
309543    9999-12-31  
309544    9999-12-31  
309545    9999-12-31  

[309546 rows x 6 columns]

## Key Fields in the REG301_EVENT_DATA Table

### ID (Primary Key) 
The ID field serves as a technical identifier that uniquely connects patent applications across various tables.

### EVENT_DATE (Primary Key)
The ``EVENT_DATE`` attribute in the ``REG301_EVENT_DATA`` and ``REG731_EVENT_DATA`` tables refers to the date when the event was recorded in the file. This date may differ from the effective legal or filing date of the event itself, as it marks when the action was officially entered into the system. The ``EVENT_DATE`` helps track the timing of the event within the EPO's records, providing an important reference point for understanding the sequence of events in the patent application process.

### The EVENT_CODE (Primary Key)
This attribute in the ``REG301_EVENT_DATA`` table represents the internal code used by the EPO to identify the type of legal action or event related to a patent application. This code is essential for classifying and tracking the various actions, such as publications, changes, or deletions, that occur throughout the patent process. The code is up to 30 characters long and may include descriptive elements, such as "9199" for actions before B1 publication or "9299" for actions after B1 publication. It helps ensure consistent identification and understanding of each specific event.

In [4]:
q = db.query(
    REG301_EVENT_DATA.event_code
).distinct()

res = patstat.df(q)
res


event_code
0            0009182
1            0009013
2        0009199INVT
3        0009299OPPO
4            0009260
..               ...
319      EPIDOSNIGR7
320      EPIDOSNRFE2
321      EPIDOSNWIAI
322    EPIDOSNEXAM52
323  EPIDOSNPFE2RFEE

[324 rows x 1 columns]

### BULLETIN_YEAR 
In the PATSTAT database, the ``BULLETIN_YEAR`` field captures the year when an action or event related to a patent application was published in the EPO Bulletin. This field plays a critical role in tracking the timeline of patent events, ensuring chronological accuracy in analyses.

The ``BULLETIN_YEAR`` is a 4-digit numeric field (formatted as YYYY), with a default value of 0 to indicate cases where no bulletin publication is known. For entries where publication in the EPO Bulletin is confirmed, ``BULLETIN_YEAR`` reflects the corresponding year of publication. It is used in conjunction with ``BULLETIN_NR``, which specifies the bulletin issue number.

The attributes ``BULLETIN_YEAR`` and ``BULLETIN_NR`` always hold non-default values in tables such as:

- REG113_TERMS_OF_GRANT: Documents terms under which patents are granted.
- REG118_PREV_FILED_APPLN: Tracks previously filed applications.
  
### BULLETIN_NR

The ``BULLETIN_NR`` attribute represents the issue number of the EPO Bulletin in which a specific action has been published. This number indicates the calendar week during which the Bulletin was released. It serves as a reference for identifying the exact edition of the EPO Bulletin where actions such as patent grants, publications, or other significant events are announced.

If the action was not published in the Bulletin or if the information is unknown, the default value of 0 is used for the ``BULLETIN_NR``, which corresponds to the absence of a known bulletin number. This value is only used when the associated ``BULLETIN_YEAR`` is also set to 0.

### BULLETIN_DATE
This attribute in the ``REG301_EVENT_DATA`` and ``REG731_EVENT_DATA`` tables refers to the date when the event was published in the bulletin. For events that are also published in the official gazette or bulletin, this date marks the public disclosure of the event. If the event has not been published in the bulletin, the default value is set to 9999-12-31. This attribute provides a crucial reference for understanding when legal events are communicated publicly, helping to track the dissemination of patent-related information.